presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -648,7 +648,7 @@ private Optional<List<String>> doGetPartitionNames(
                     partitionNameResult = Optional.of(delegate.getPartitionNamesByFilter(metastoreContext, hiveTableHandle.getSchemaName(), hiveTableHandle.getTableName(), partitionPredicates));
                 }
                 else {
-                    partitionNameResult = delegate.getPartitionNames(metastoreContext, hiveTableHandle.getSchemaName(), hiveTableHandle.getTableName());
+                    partitionNameResult = delegate.getPartitionNames(metastoreContext, table);
```

Comment:
we know the table exists .. since isPresent check is already made in line 636. we can send table.get() as input.
Having an optional value for the main parameter of getPartitionNames doesnt make sense

Reply:
The problem is getPartitionNames gets called from getPartitionNamesByFilter as well. If I change the method signature of getPartitionNames to remove Optional<Table> and have Table instead, that effectively shifts the empty table checks to other places, which is not desirable. Hence I would prefer keeping it as is.